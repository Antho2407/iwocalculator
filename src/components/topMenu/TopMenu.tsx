import React from "react";
import { TopMenuRoot } from "./TopMenu.components";
import FieldRow from "../fieldRow/FieldRow";
import { Input } from "../input/Input.components";

export interface TopMenuProps {
  onAmountChanged: (value: number) => void;
  onDurationChanged: (value: number) => void;
  amount: number;
  duration: number;
}

const TopMenu: React.FC<TopMenuProps> = ({
  onAmountChanged,
  onDurationChanged,
  amount,
  duration,
}) => {
  const handleAmountChanged = (e: React.ChangeEvent<HTMLInputElement>) => {
    onAmountChanged(parseInt(e.target.value));
  };
  const handleDurationChanged = (e: React.ChangeEvent<HTMLInputElement>) => {
    onDurationChanged(parseInt(e.target.value));
  };

  return (
    <TopMenuRoot>
      <FieldRow label="Amount requested" info="(in Â£)">
        <Input type="number" onChange={handleAmountChanged} value={amount} />
      </FieldRow>
      <FieldRow label="Amount requested" info="(in months)">
        <Input
          type="number"
          min={0}
          step={1}
          onChange={handleDurationChanged}
          value={duration}
        />
      </FieldRow>
    </TopMenuRoot>
  );
};

export default TopMenu;
