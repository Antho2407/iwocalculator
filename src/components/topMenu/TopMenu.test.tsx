import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import TopMenu, { TopMenuProps } from "./TopMenu";

describe("<TopMenu />", () => {
  let wrapper: ShallowWrapper<TopMenuProps>;

  describe("When the component is mounted", () => {
    it("should render successfully", () => {
      wrapper = shallow(
        <TopMenu
          onAmountChanged={() => {}}
          onDurationChanged={() => {}}
          amount={1000}
          duration={4}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
