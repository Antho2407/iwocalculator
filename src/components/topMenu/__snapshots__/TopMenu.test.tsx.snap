// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopMenu /> When the component is mounted should render successfully 1`] = `ShallowWrapper {}`;
